File: cpp/src/mavsdk/core/autopilot.cpp
#include "autopilot.h"

namespace mavsdk {

std::ostream& operator<<(std::ostream& str, const Autopilot& autopilot)
{
    switch (autopilot) {
        case Autopilot::Px4:
            return str << "Px4";
        case Autopilot::ArduPilot:
            return str << "ArduPilot";
        default:
            return str << "Unknown";
    }
}

} // namespace mavsdk

File: kt/mavsdk-kotlin/src/commonMain/kotlin/io/mavsdk/kotlin/Autopilot.kt
package io.mavsdk.kotlin

/** Autopilot types */
enum class Autopilot(val value: Int) {
    UNKNOWN(0),
    PX4(1),
    ARDUPILOT(2),
    GENERIC(3);

    companion object {
        fun fromValue(value: Int): Autopilot = entries.find { it.value == value } ?: UNKNOWN
    }
}
